### PR TITLE
lib: json: add helper macro for array of array

### PR DESCRIPTION
**Pull request by otavio**
The new JSON_OBJ_DESCR_ARRAY_ARRAY allows use of an array of array.

The patch has been kindled provided by Leandro Pereira
<leandro.pereira@intel.com>.

Signed-off-by: Otavio Salvador <otavio@ossystems.com.br>

### CODE REVIEW

**Comment by otavio** [comment]
@lpereira could you take a look at this?

File: include/json.h [hunk 1]
```diff
@@ -274,6 +274,75 @@ typedef int (*json_append_bytes_t)(const char *bytes, size_t len,
...
+ *          JSON_OBJ_DESCR_OBJECT(struct person_heigths_array, heights, person_heigth_descr),
+ *      };
+ *
+ *      struct json_obj_descr array_array[] = {
+ *           JSON_OBJ_DESCR_ARRAY_ARRAY(struct people_heights, heights, 10,
+ *                                      heights_len, person_height_array_descr,
+ *                                      ARRAY_SIZE(person_height_array_descr)),
+ *      };
+ */
+#define JSON_OBJ_DESCR_ARRAY_ARRAY(struct_, field_name_, max_len_, len_field_, \
```
Comment (nashif) [inline include/json.h]: can you please add a test for this new macro?

**Review by nashif — DISMISSED** [review]

File: include/json.h (hunk 1 above)
Comment (otavio) [reply include/json.h]: sure, we will work on it.

**Comment by otavio** [comment]
@nashif, about the commit log, do you have any suggestion?

**Review by lpereira — APPROVED** [review]
LGTM, thanks for sending this in.  Please add a test like @nashif suggested.  The samples provided by @chtavares592 in #8567 are more than sufficient.

**Review by lpereira — APPROVED** [review]

**Comment by lpereira** [comment]
Patch LGTM, but please [fix the style issue](https://app.shippable.com/github/zephyrproject-rtos/zephyr/runs/18519/5/tests) that is preventing CI from being completely happy.

**Comment by otavio** [comment]
@lpereira it should be good to go now. Please double check the commit log :-)

**Review by nashif — APPROVED** [review]

**Pull request merged**